### PR TITLE
EOS-17038: cortx-ha needs an interface to query motr process fids

### PR DESCRIPTION
**Pull request by Shreya-18**
### Description:  
jira ticket- https://jts.seagate.com/browse/EOS-17038
Fix for issue: https://jts.seagate.com/browse/EOS-17038?focusedCommentId=2068146&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2068146
1. For a specified service on a given node, return all the service fids matching the given service.
2. Handling FileNotFoundError exception

### Solution:
1. Returning list of service fids matching the given service.
2. Logging message for FileNotFoundError exception.

### OUTPUT:
```
$ hctl fetch-fids --help
usage: hare-fetch-fids [OPTION]

Fetches the fids for services configured on a node.

optional arguments:
  -h, --help            show this help message and exit
  --service SERVICE, -s SERVICE
                        service name. - Returns fid for given service. List of
                        services- "confd", "ioservice", "s3server". Default:
                        Returns the fids for all the services.
  --node NODE, -n NODE  node-name - Returns the fids of services for the given
                        node. Default: Local node
  --all                 Returns fids of all the services for nodes in cluster.
  --json                show output in JSON format

```
```
$ hctl fetch-fids -n ssc-vm-c-1813.colo.seagate.com --json
[
  {
    "name": "confd",
    "fid": "0x7200000000000001:0x9"
  },
  {
    "name": "ioservice",
    "fid": "0x7200000000000001:0xc"
  },
  {
    "name": "ioservice",
    "fid": "0x7200000000000001:0x1d"
  }
]
```
```
$ hctl fetch-fids -n ssc-vm-c-1813.colo.seagate.com -s ioservice --json
[
  {
    "name": "ioservice",
    "fid": "0x7200000000000001:0xc"
  },
  {
    "name": "ioservice",
    "fid": "0x7200000000000001:0x1d"
  }
]
```

```
$ hctl fetch-fids --all --json
[
  {
    "name": "ssc-vm-c-1813.colo.seagate.com",
    "svcs": [
      {
        "name": "confd",
        "fid": "0x7200000000000001:0x9"
      },
      {
        "name": "ioservice",
        "fid": "0x7200000000000001:0xc"
      },
      {
        "name": "ioservice",
        "fid": "0x7200000000000001:0x1d"
      }
    ]
  },
  {
    "name": "ssc-vm-c-1898.colo.seagate.com",
    "svcs": [
      {
        "name": "confd",
        "fid": "0x7200000000000001:0x35"
      },
      {
        "name": "ioservice",
        "fid": "0x7200000000000001:0x38"
      }
    ]
  }
]
```
For running script from other than primary node (non-bootstrapping node) that does not contain the `consul-kv.json` file.
```
$ hctl fetch-fids -n ssc-vm-c-1898.colo.seagate.com
[Errno 2] No such file or directory: '/var/lib/hare/consul-kv.json'
```

### CODE REVIEW

File: rfc/11/README.md [hunk 1]
```diff
@@ -19,13 +19,17 @@ Interact with Motr cluster.
 Commands:
 
     bootstrap   bootstrap the cluster
+    drive-state post drive event to Motr
+    fetch-fids  Fetch the fids for services configured on a node
```
Comment (suvratjoshi) [inline rfc/11/README.md:23]: Would be nice to add which fids we are fetching motr, s3 etc

Comment (Shreya-18) [reply rfc/11/README.md]: Done

**Review by suvratjoshi — APPROVED** [review]

**Review by knekrasov — CHANGES_REQUESTED** [review]
When I run mypy against this file, I can see errors:

```
MYPYPATH=./stubs:./hax:./utils/ mypy --config-file hax/mypy.ini utils/hare-fetch-fids
utils/hare-fetch-fids:35: error: Need type annotation for 'svcs' (hint: "svcs: List[<type>] = ...")
utils/hare-fetch-fids:105: error: Missing return statement
utils/hare-fetch-fids:119: error: Returning Any from function declared to return "List[Process]"
utils/hare-fetch-fids:119: error: Argument 1 to "get_service" of "Node" has incompatible type "Optional[str]"; expected "str"
utils/hare-fetch-fids:158: error: Argument 1 to "Node" has incompatible type "Node"; expected "str"
Found 5 errors in 1 file (checked 1 source file)
```

Side note: it seems like our Makefile should be fixed, it doesn't check this file with mypy. Although it should.

**Comment by Shreya-18** [comment]
> When I run mypy against this file, I can see errors:
> 
> ```
> MYPYPATH=./stubs:./hax:./utils/ mypy --config-file hax/mypy.ini utils/hare-fetch-fids
> utils/hare-fetch-fids:35: error: Need type annotation for 'svcs' (hint: "svcs: List[<type>] = ...")
> utils/hare-fetch-fids:105: error: Missing return statement
> utils/hare-fetch-fids:119: error: Returning Any from function declared to return "List[Process]"
> utils/hare-fetch-fids:119: error: Argument 1 to "get_service" of "Node" has incompatible type "Optional[str]"; expected "str"
> utils/hare-fetch-fids:158: error: Argument 1 to "Node" has incompatible type "Node"; expected "str"
> Found 5 errors in 1 file (checked 1 source file)
> ```
> 
> Side note: it seems like our Makefile should be fixed, it doesn't check this file with mypy. Although it should.

```
$ MYPYPATH=./stubs:./hax:./utils/ mypy --config-file hax/mypy.ini utils/hare-fetch-fids
Success: no issues found in 1 source file

```
Fixed all issues.

File: utils/hare-fetch-fids [hunk 2]
```diff
@@ -106,16 +112,19 @@ def get_service_for_node(nodes: List[Node], node_name: str = None, service: str
...
 
+    # returns all the services for the specified given service
     for node in nodes:
         if node.name == node_name:
             return node.get_service(service)
 
-    print(f'Node {node_name} not found in cluster.', file=sys.stderr)
+    print(
+        f'Node {node_name} not found in cluster.', file=sys.stderr)
+    return None
```
Comment (knekrasov) [inline utils/hare-fetch-fids:124]: [Optional]: usually it is not the best practice to return None; this forces the caller to make additional check (if not None is returned) - more code to write and one more point where a mistake can happen.

In other words, returned value here aims two different things: return the result of some computation AND report about an error. But why do we need the latter if the language supports exceptions? You can throw an exception if something unexpected is happening. Otherwirse return the actual value.

But I don't require it for this script, I think we can merge it as it is now.

**Review by knekrasov — APPROVED** [review]

File: utils/hare-fetch-fids (hunk 2 above)
Comment (Shreya-18) [reply utils/hare-fetch-fids:124]: Oh yes, thanks a lot!

> You can throw an exception if something unexpected is happening. Otherwirse return the actual value.
Makes sense. Will keep in mind.

**Pull request merged**